### PR TITLE
PP-590: Correcting PTL to fail when MoM other than MS sends resources_used info to the server

### PR DESCRIPTION
**Pull request by prakashcv13**
#### Issue-ID

* **[PP-590](https://pbspro.atlassian.net/browse/PP-590)**

#### Problem description
* The PTL test cases written earlier through [PR-228](https://github.com/PBSPro/pbspro/pull/228) were passing even on builds without the fix.

#### Cause / Analysis
* The test case was not checking for resources_used information in qstat output. 

#### Solution description
* Added the check to test for resources_used in qstat o/p.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by dilip-krishnan** [comment]
Hi Prakash,
    Why are we just checking ncpus resource, there are other resources in the bug ticket , which were reported showing 0 value as part of the bug.,like cpupercent,cput, mem and vmem   We should validate these resources as well.

File: test/tests/functional/pbs_two_mom_hooks_resources_used.py [hunk 1]
```diff
@@ -103,6 +104,8 @@ def test_Rrecord(self):
         # Wait for the job to start running.
         self.server.expect(JOB, {ATTR_substate: '42'}, jid1)
```
Comment (prakashcv13) [inline test/tests/functional/pbs_two_mom_hooks_resources_used.py]: I prefer checking for substate.

File: test/tests/functional/pbs_two_mom_hooks_resources_used.py [hunk 2]
```diff
@@ -92,7 +93,7 @@ def test_Rrecord(self):
         test = []
         test += ['#PBS -N NodeFailRequeueTest\n']
         test += ['echo Starting test at `date`\n']
-        test += ['sleep 20\n']
+        test += ['sleep 100\n']
```
Comment (prakashcv13) [inline test/tests/functional/pbs_two_mom_hooks_resources_used.py]: changed.

File: test/tests/functional/pbs_two_mom_hooks_resources_used.py [hunk 3]
```diff
@@ -103,6 +104,8 @@ def test_Rrecord(self):
         # Wait for the job to start running.
         self.server.expect(JOB, {ATTR_substate: '42'}, jid1)
         # Kill the MoM process on the MS.
+
+        time.sleep(30)
```
Comment (prakashcv13) [inline test/tests/functional/pbs_two_mom_hooks_resources_used.py]: there is no need of this sleep in fact, so removed it.

**Comment by prakashcv13** [comment]
@bhroam, I have addressed your comments, please review.

File: test/tests/functional/pbs_two_mom_hooks_resources_used.py (hunk 1 above)
Comment (bhroam) [inline test/tests/functional/pbs_two_mom_hooks_resources_used.py]: I'd like to remain consistent with other tests.  I've seen tests check for state: 'R' or both state and substate.  The latter being during a period before we fixed expect to check for substate itself.  It's not needed any more.  I've not seen tests just check for substate before.

**Comment by prakashcv13** [comment]
@bhroam, I have changed the script to check for state to be 'R', instead of substate to be '42'.

**Comment by bhroam** [comment]
Tanks for making the changes.  Looks good to me.  I sign off on the code.  It looks like Travis is having an issue though.  The log says checksign failed, but I don't understand why.  You're commits are signed.

**Comment by bhroam** [comment]
Oh, @prakashcv13, can you update the pull request title as I suggested?  Referring to "the fix" isn't very explanatory.

**Comment by prakashcv13** [comment]
@bhroam, I have updated the title.. please let me know if this one works for you..

**Comment by bhroam** [comment]
@prakashcv13 thanks for updating the title.  It looks good now.

**Comment by arungrover** [comment]
Still looks good to me. Please rebase your branch

**Comment by iestockdale** [comment]
Bhroam, Praskash, note that checking for the 'R' state alone leaves tests prone to false failures due to race conditions.  It is strongly advised to always check for substate 42.

**Comment by bhroam** [comment]
Ian, we no longer need to check for substate 42.  PTL does that for us.  There is no race condition.

**Comment by iestockdale** [comment]
Bhroam, thanks for adding that information.

**Comment by prakashcv13** [comment]
@arungrover, @bhroam, I have rebased my branch. Please take it forward...
@iestockdale, Checking for substate is no longer necessary after [PP-513](https://pbspro.atlassian.net/browse/PP-513) which was implemented through this [PR](https://github.com/PBSPro/pbspro/pull/196)

**Comment by prakashcv13** [comment]
@arungrover, @bhroam, there was an issue with my previous rebase.. I have corrected it now.. Could you please take it forward..

**Comment by arungrover** [comment]
Merged

**Comment by arungrover** [comment]
@prakashcv13 I just realized that this pull had a open comment from @dilip-krishnan.
Can you please reply back to him with your findings or else fix it if you guys think that it is legitimate.

**Comment by prakashcv13** [comment]
@dilip-krishnan, I am not clear if the comment regarding checking for other resources is for which test case. However, as per the fix, it is sufficient to check only for the walltime in test_Erecord. The merged content of the tests is enough for verifying the fix and checking for other resources will not be required as
this version of the tests has been proven enough for verification on other branches. 
Also, as far as I know, implementer of the change needs to provide basic testing as part of the fix which I have already done and according to me checking for other resources will be "extra", which if anyone feels should be done for regression testing or any other reason is free to contribute.

**Pull request merged**